cpp/src/arrow/util/rle_encoding_test.cc

```diff
@@ -227,10 +234,9 @@ void ValidateRle(const std::vector<int>& values, int bit_width,
   if (expected_len != -1) {
     EXPECT_EQ(encoded_len, expected_len);
   }
-  if (expected_encoding != NULL) {
+  if (expected_encoding != NULL && encoded_len == expected_len) {
```

Comment:
Why do we need this?

Reply:
that's a bug in the test which I came across - because it uses `EXPECT_EQ`  as opposed to `ASSERT_EQ` above when checking the two lengths are the same , then whilst the test fails, the function continues to run - so in the case an incorrect length is returned, `memcmp` still runs and is comparing beyond the length of one of the buffers, and causing the whole test process to crash.

Reply:
OK. Let's change the above `EXPECT_EQ()` to `ASSERT_EQ()` and remove the added condition.